source/Core/InputValidator.php

```diff
@@ -186,14 +186,26 @@ public function checkEmail($user, $email)
         }
 
         // invalid email address ?
-        if (!oxNew(\OxidEsales\Eshop\Core\MailValidator::class)->isValidEmail($email)) {
+        if (!$this->isValidEmail($email)) {
             $exception = oxNew(\OxidEsales\Eshop\Core\Exception\InputException::class);
             $exception->setMessage(\OxidEsales\Eshop\Core\Registry::getLang()->translateString('ERROR_MESSAGE_INPUT_NOVALIDEMAIL'));
 
             return $this->addValidationError("oxuser__oxusername", $exception);
         }
     }
 
+    /**
+     * @param strgin $email
+     * @return bool
+     */
+    public function isValidEmail($email)
+    {
+        if (!is_string($email) || !strlen($email)) {
+            return false;
+        }
+        return filter_var($email, FILTER_VALIDATE_EMAIL) !== false;
```

Comment:
Maybe use this instead:
```php
return filter_var($email, FILTER_VALIDATE_EMAIL) 
  && preg_match('/@.+\./', $email);
```
... and add a test for `abc@def` as well

Reply:
see https://github.com/OXID-eSales/oxideshop_ce/pull/725#discussion_r335406376

Reply:
Ok, I rechecked that and that shouldn't be a problem since PHP 5.3.3 (which is really old...)
![image](https://user-images.githubusercontent.com/1001186/66914735-c1550780-f017-11e9-87bd-e8f13e3861c7.png)